src/Elastic.Documentation.LinkIndex/LinkIndexProvider.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System.Net;
+using Amazon.Runtime;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Elastic.Documentation.Links;
+
+namespace Elastic.Documentation.LinkIndex;
+
+public interface ILinkIndexProvider
```

Comment:
Is it worth splitting this into `ILinkIndexReadProvider` and have `ILinkIndexProvider` inherit from that? 

Most usages do not need access to the `Save()` operation.

Reply:
Thank you for the nice suggestions. Unfortunately, I did all in one commit.

https://github.com/elastic/docs-builder/pull/1302/commits/18309d9ce9be51fb5576b1e0d47d703725afddfa